### PR TITLE
FIX sale multi-company

### PR DESCRIPTION
**Pull request by Feyensv**
Task ID: 1986256

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale_crm/models/crm_lead.py [hunk 1]
```diff
@@ -80,5 +80,7 @@ def action_new_quotation(self):
             'default_origin': self.name,
             'default_name': self.name,
             'default_source_id': self.source_id.id
+            # 'default_company_id': will be set to self.env.company_id
```
Comment (dbo-odoo) [inline addons/sale_crm/models/crm_lead.py]: useless comment

File: addons/sale_management/models/sale_order.py [hunk 2]
```diff
@@ -210,5 +210,6 @@ def _get_values_to_add_to_order(self):
             'product_uom_qty': self.quantity,
             'product_uom': self.uom_id.id,
             'discount': self.discount,
-            'company_id': self.order_id.company_id.id,
+            # 'company_id': self.order_id.company_id.id,
```
Comment (dbo-odoo) [inline addons/sale_management/models/sale_order.py]: useless comment

File: addons/sale_margin/models/sale_order.py [hunk 3]
```diff
@@ -30,7 +30,7 @@ def _get_purchase_price(self, pricelist, product, product_uom, date):
             purchase_price = product.uom_id._compute_price(purchase_price, product_uom)
         price = frm_cur._convert(
             purchase_price, to_cur,
-            self.order_id.company_id or self.env.company_id,
+            self.company_id or self.env.company_id,
```
Comment (dbo-odoo) [inline addons/sale_margin/models/sale_order.py]: `self.company_id or self.env.company`

File: addons/sale_margin/models/sale_order.py [hunk 4]
```diff
@@ -70,5 +70,5 @@ class SaleOrder(models.Model):
 
     @api.depends('order_line.margin')
     def _product_margin(self):
-        for order in self:
-            order.margin = sum(order.order_line.filtered(lambda r: r.state != 'cancel').mapped('margin'))
+        for order in self.filtered(lambda order: order.state != 'cancel'):
```
Comment (dbo-odoo) [inline addons/sale_margin/models/sale_order.py]: i know it's difficult to resist, but keep your commits within the scope of the task plz
this has nothing to do with multi-company

Comment (Feyensv) [reply addons/sale_margin/models/sale_order.py]: It came from the order.state/order.company_id removal, but I can drop it if you want :P

**Comment by dbo-odoo** [comment]
closed since tivisse already did most of it, the rest is mostly noise at this stage

sowwy

**Pull request closed without merging**